### ISSUE TITLE
Question: score too slow but it should be higher

### ISSUE THREAD

**Issue by fernandog** [Diaoul/subliminal#586]
I'm doing some tests and I don't understand why this sub is not downloaded for file: Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv

http://www.opensubtitles.org/pt/subtitles/6498938/agent-carter-the-atomic-job-pb

This shouldn't match because of the '/' in the name:
`Marvel\'s.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD.H.264-Coo7.mkv`

but this should download:
`Marvels.Agent.Carter.S02E05.The.Atomic.Job.720p.WEB-DL.DD5.1.H.264-Coo7.mkv` isn't?

UPDATE:
Its because serie name in subtitle misses the single quote?
Can you match series name with sanitize title and un-sanitize title?
this happens in LTV too

Hash is not calculate because it's a dummy file

The min score of 213 in SR is:
```
Perfect match = hash score - hearing impaired score - resolution score (subtitle for 720p is the same as for 1080p)
    # Perfect match = 215 - 1 - 1 = 213
```

```
2016-02-19 13:55:44 DEBUG    FINDSUBTITLES :: [opensubtitles] Subtitle score for 1955081202 is: 108 (min=213)
AA2016-02-19 13:55:44 INFO     Score 108 is below min_score (213)
AA2016-02-19 13:55:44 INFO     Computed score 108 with final matches set(['episode', 'title', 'season', 'format', 'hearing_impaired', 'audio_codec', 'year', 'video_codec', 'resolution', 'release_group'])
AA2016-02-19 13:55:44 DEBUG    Matched hearing_impaired
AA2016-02-19 13:55:44 DEBUG    Adding title match equivalent
AA2016-02-19 13:55:44 DEBUG    Found matches set(['episode', 'title', 'season', 'format', 'audio_codec', 'year', 'video_codec', 'resolution', 'release_group'])
AA2016-02-19 13:55:44 DEBUG    Using scores {'hash': 215, 'format': 4, 'series': 108, 'audio_codec': 2, 'year': 54, 'hearing_impaired': 1, 'episode': 18, 'season': 18, 'video_codec': 1, 'resolution': 1, 'release_group': 9}
AA2016-02-19 13:55:44 INFO     Computing score of  for video  with {'hearing_impaired': False}
2016-02-19 13:55:44 INFO     FINDSUBTITLES :: Found subtitle(s) canditate(s) for Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv
AA2016-02-19 13:55:44 DEBUG    Found subtitle 
AA2016-02-19 13:55:43 INFO     Searching subtitles [{'sublanguageid': u'pob', 'tag': "Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv"}, {'query': 'Marvels Agent Carter', 'sublanguageid': u'pob', 'episode': 5, 'season': 2}]
AA2016-02-19 13:55:43 DEBUG    Logged in with token '5kfgept3f49k7u5kmvbdtb1g84'
AA2016-02-19 13:55:42 INFO     Logging in
AA2016-02-19 13:55:42 INFO     Initializing provider opensubtitles
AA2016-02-19 13:55:42 INFO     Listing subtitles with provider 'opensubtitles' and languages set([])
AA2016-02-19 13:55:42 DEBUG    Logged in
AA2016-02-19 13:55:40 INFO     Logging in
AA2016-02-19 13:55:40 INFO     Initializing provider legendastv
AA2016-02-19 13:55:40 INFO     Listing subtitles with provider 'legendastv' and languages set([])
AAMalformedMKVError: No Segment found
AA    raise MalformedMKVError('No Segment found')
AA  File "/home/osmc/SickRage/lib/enzyme/mkv.py", line 42, in __init__
AA    mkv = MKV(f)
AA  File "/home/osmc/SickRage/lib/subliminal/video.py", line 316, in scan_video
AATraceback (most recent call last):
AA2016-02-19 13:55:40 ERROR    Parsing video metadata with enzyme failed
AA2016-02-19 13:55:40 WARNING  Size is lower than 10MB: hashes not computed
AA2016-02-19 13:55:39 INFO     Scanning video "Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv"
```

**Comment by Diaoul** [comment]
Please reproduce with latest develop and debug logs.

**Comment by fernandog** [comment]
we are using latest subliminal (develop in SR (develop)
don't know if is related but:

Passing: 
```python
video_filename = "/downloaded/series/Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv"
video = subliminal.scan_video(video_filename,subtitles=False, embedded_subtitles=False)
```

results in:
```
Scanning video "Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv" in '/media/SAMSUNG/media/downloaded/series'
Subliminal video: <Episode ["Marvel's Agent Carter", 2x5]>
```

BUT
passing:
```python
video_filename = "/downloaded/series/Marvels.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H264-Coo7[rartv]/Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv"
video = subliminal.scan_video(video_filename,subtitles=False, embedded_subtitles=False)
```

```
Scanning video "Marvel's.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H.264-Coo7.mkv" in '/downloaded/series/Marvels.Agent.Carter.S02E05.The.Atomic.Job.1080p.WEB-DL.DD5.1.H264-Coo7[rartv]'
Subliminal video: <Episode ['Marvels Agent Carter', 2x5]>
```

the only difference is that the second one has a folder with the same name (almost)  as the video file
Folder is: Marvels.Agent
File is: Marvel's.Agent

**Comment by Diaoul** [comment]
The reason why subliminal finds series 'Marvels Agent Carter' and not 'Marvel's Agent Carter' for the second video is because of the path. There is no way for guessit to know which one is the correct one and I think in case of conflicts the name of the video matters more than the name of the folder (which is pretty legitimate).

Your logs seem striped, please paste CLI debug logs on pastebin or similar.

**Comment by fernandog** [comment]
Will do. Away this weekend.
Should I open an issue in guessit?
It should get filename and strip any folder isnt? If fails, get foldet name

**Comment by h3llrais3r** [comment]
Filename normally should have precedence over folder name.
This has been implemented in guessit 0.11.
See https://github.com/guessit-io/guessit/issues/190.

**Comment by Diaoul** [comment]
I concur, as long as the filename carries more information that is.

**Comment by fernandog** [comment]
@Diaoul here is the CLI log: https://gist.github.com/fernandog/328435f409ebd2c539dd

the error in the end doesn't happen in SR (or at least it's not showed)

**Comment by Diaoul** [comment]
@fernandog: use develop branch of subliminal and up-to-date dependencies please.

**Comment by fernandog** [comment]
I did: sudo pip install https://github.com/Diaoul/subliminal/archive/develop.zip

updated gists. check if they are ok now.
thanks!

**Comment by Diaoul** [comment]
You still have an old guessit version it seems.

**Comment by fernandog** [comment]
```
10:00:43 osmc@osmc:~/SickRage (develop)$ guessit --version
+-------------------------------------------------------+
+                   GuessIt 2.0.4                       +
+-------------------------------------------------------+
|      Please report any bug or feature request at      |
|       https://github.com/wackou/guessit/issues.       |
+-------------------------------------------------------+
10:00:49 osmc@osmc:~/SickRage (develop)$
```

**Comment by Diaoul** [comment]
Your logs show guessit transformers and there are no more transformers in guessit 2.
Subliminal version is 0.7.4 : https://gist.github.com/fernandog/c78ded72829588a160e9#file-gistfile1-txt-L79

**Comment by fernandog** [comment]
figure it out. pip installed in  /usr/local/bin/

gists updated
sorry!

**Comment by Diaoul** [comment]
Put a `print` [here](https://github.com/Diaoul/subliminal/blob/develop/subliminal/providers/opensubtitles.py#L57) of the results of both sanitize. This should help.

**Comment by fernandog** [comment]
like this https://gist.github.com/fernandog/452ec67de253bdfb0eb9 ?

```python
            logger.info('Unsanitize name: %s', self.series_name)
            logger.info('Sanitize name: %s', sanitize(self.series_name))
            logger.info('Unsanitize video: %s', video.series)
            logger.info('Sanitize video: %s', sanitize(video.series))
```

http://www.opensubtitles.org/en/search/sublanguageid-pob/imdbid-4706432

but in OS show name is "Agent Carter" and in filename is "Marvel's Agent Carter", so will never match even comparing with unsanitized strings or not ? it found the subs even if names are different. I didn't know it matched series name in OS.

I thought it only get the subttile filename

in this case it should be a exact match:
`Marvels.Agent.Carter.S02E05.The.Atomic.Job.720p.WEB-DL.DD5.1.H.264-Coo7.mkv`
http://www.opensubtitles.org/en/subtitles/6498938/agent-carter-the-atomic-job-pb

**Comment by fernandog** [comment]
about OS:

Not sure how to fix as the only way it could download is changing opensubtitles.py and do:
`sanitize(self.series_name) in sanitize(video.series)`
but I'm afraid it will only work for this case. if OS has a full name and file name a short name it wont work

```python
            if video.series and (sanitize(self.series_name) == sanitize(video.series) or sanitize(self.series_name) in sanitize(video.series)):
                matches.add('series')
```


about LTV issue not matching because of the single quote:

I think this can be done. Match sanitize or unsanitize (changing subtitles.py)

```python
    matches = set()
    if isinstance(video, Episode):
        # series
        if video.series and 'series' in guess and (sanitize(guess['series']) == sanitize(video.series) or guess['series'] == video.series):
```

**Comment by Diaoul** [comment]
Fixed with b5b6dc827a9972e089ba9f9b54f88c5893ce722b and c3c0c4584f14ebafb59381a6ca96e3b20c5cc799
